agent/hcp/bootstrap/bootstrap.go

```diff
@@ -443,3 +453,25 @@ func ValidateTLSCerts(cert, key string, caCerts []string) error {
 	}
 	return nil
 }
+
+// LoadManagementToken returns the management token, either by loading it from the persisted
+// token config file or by fetching it from HCP if the token file does not exist.
+func LoadManagementToken(ctx context.Context, logger hclog.Logger, client hcpclient.Client, dataDir string) (string, error) {
+	hcpCfgDir := filepath.Join(dataDir, SubDir)
+	token, err := loadManagementToken(hcpCfgDir)
+
+	if err != nil {
+		logger.Debug("fetching configuration from HCP")
+		var err error
+		cfg, err := fetchBootstrapConfig(ctx, client, dataDir)
+		if err != nil {
+			return "", err
+		}
+		logger.Debug("configuration fetched from HCP and saved on local disk")
+		token = cfg.ManagementToken
+	} else {
+		logger.Trace("loaded HCP configuration from local disk")
```

Comment:
Why is this logged at `Trace` level but the others are `Debug`? Just curious because they seem like the same level of granularity to me

Reply:
Mostly because we go through this code path more often than the other one! We typically will fetch and save once and then all subsequent calls of this method will load from local disk. Since the reconcile method calls this load method and reconcile can happen multiple times, it felt less spammy to set this to trace.